File: C++/Question 3.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

class node {
    public:
        int data;
        node* left;
        node* right;

        node(int value) : data(value), left(nullptr), right(nullptr) {}
};

int sumNoChildren(node* root) {
    if (root == nullptr) {
        return 0;
    } else if (root->left == nullptr && root->right == nullptr) {
        return root->data;
    }
    return sumNoChildren(root->left) + sumNoChildren(root->right);
}

int main() {
    int n;
    cin >> n;
    n -= 1;
    node* root = new node(0);
    int initial_root;
    cin >> initial_root;
    root->data = initial_root;

    for (int i = 0; i < n; i++) {
        int data;
        cin >> data;
        node* current = root;
        while (true) {
            if (data <= current->data) {
                if (current->left == nullptr) {
                    node* n = new node(0);
                    n->data = data;
                    current->left = n;
                    break;
                } else {
                    current = current->left;
                }
            } else {
                if (current->right == nullptr) {
                    current->right = new node(data);
                    break;
                } else {
                    current = current->right;
                }
            }
        }
    }

    cout << sumNoChildren(root) << endl;

    return 0;
}

File: Java/Question3.java
import java.util.Scanner;

public class Question3 {
	static class Node {
		final int data;
		Node left, right;

		Node(int d) {
			data = d;
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		Node root = new Node(sc.nextInt());

		for(int i = 1; i < n; i++) {
			int data = sc.nextInt();
			Node c = root;
			while(true) {
				if(data <= c.data) {
					if(c.left == null) {
						c.left = new Node(data);
						break;
					} else {
						c = c.left;
					}
				} else {
					if(c.right == null) {
						c.right = new Node(data);
						break;
					} else {
						c = c.right;
					}
				}
			}
		}

		System.out.println(countNoChildren(root));
	}

	static int countNoChildren(Node root) {
		if(root == null) return 0;
		if(root.left == root.right) return root.data;
		return countNoChildren(root.left) + countNoChildren(root.right);
	}
}
